### ISSUE TITLE
debian repository for virtuoso 7.2

### ISSUE THREAD

**Issue by Jan-Marcek** [openlink/virtuoso-opensource#350]
Is there debian repository for release 7.2?
Thank you

PS: Maybe good news. I have successfully created deb package for 7.2.

**Comment by HughWilliams** [comment]
@Jan-Marcek: I presume you used [these instructions](http://virtuoso.openlinksw.com/dataspace/doc/dav/wiki/Main/VOSDebianNotes) to create these 7.2 packages?

We hope Debian (and others) will create a set at some point for their next official distributions, but when that will happen is unknown ...

**Comment by Jan-Marcek** [comment]
Hi Hugh,
No, I didn't use the instructions at link. We really need to install the latest virtuoso via debian package. However, I haven't seen any repository for 7.2 that's why I have had to build debian packages. We will provide virtuoso-7.2 in our repository until you place .deb somewhere. 

We are using debian wheezy.

I used these steps:
Prerequisites

needs automake = 1.14

```
#add testing debian repository to source list
echo "deb http://ftp.au.debian.org/debian testing main" >> /etc/apt/sources.list
apt-get update
apt-get install automake -V
```

check automake version. We need 1.14

```
automake --version
make-1.14 to provide /usr/bin/automake (automake) in auto mode
```

get virtuoso and take v7.2.0.1

```
git clone https://github.com/openlink/virtuoso-opensource.git
git checkout v7.2.0.1
```

 needs build-deb for virtuoso

```
# go into virtuoso sources
cd virtuoso-opensource
apt-get install -y  build-essential
# install build-dep - prerequisites for virtuoso
deps=$(dpkg-checkbuilddeps 2>&1 | sed 's/.*: //' | sed 's/([^)]*)//g')
apt-get install -y $deps
```

build debian package

```
fakeroot debian/rules clean binary
```

FAQ

If you get an error message: configure: error: "Directory specified with --with-jdk4 does not exist"
then you have to install openjdk-6-jdk\

```
aptitude install openjdk-6-jdk
```

**Comment by mwjames** [comment]
> PS: Maybe good news. I have successfully created deb package for 7.2.

Would it be possible for you to post your deb package into the Personal Package Archives [0] so others can use `apt-get` via launchpad [2]?

[0] https://launchpad.net/debian/+ppas
[1] https://launchpad.net/virtuoso
[2] https://help.ubuntu.com/community/Repositories/CommandLine#Adding_Launchpad_PPA_Repositories

**Comment by openlink** [comment]
> > PS: Maybe good news. I have successfully created deb package for 7.2.
 
Good to hear.

> Would it be possible for you to post your deb package into the [Personal Package Archives](https://launchpad.net/debian/+ppas) so others can use apt-get via [launchpad](https://help.ubuntu.com/community/Repositories/CommandLine#Adding_Launchpad_PPA_Repositories)?

That's a great suggestion - many thanks, I'll look into it soon.

~Tim

Tim Haynes
Product Development Consultant
OpenLink Software
http://www.openlinksw.com/
http://twitter.com/openlink

**Comment by Jan-Marcek** [comment]
Hi,
we are using [aptly](http://www.aptly.info/) as debian repository management tool.

Virtuoso 7.2 is part of [OpenDataNode](http://www.comsode.eu/index.php/2014/06/open-data-node-what-it-is-what-it-does-what-is-next/).

```shell
echo "deb http://packages.comsode.eu/debian wheezy main" > /etc/apt/sources.list.d/odn.list
wget -O - http://packages.comsode.eu/key/odn.gpg.key | apt-key add -
apt-get update
```

**Comment by joernhees** [comment]
would still be cool if there was just an openlink PPA... any update on that?

**Comment by openlink** [comment]
On 23 October 2015 at 16:11, Jörn Hees notifications@github.com wrote:

> would still be cool if there was just an openlink PPA... any update on
> that?

Hi,

There has been some movement on this internally, but it'll be a little
while longer before we have anything to announce. Please bear with us :)

## ~Tim

Tim Haynes
Product Development Consultant
OpenLink Software
http://www.openlinksw.com/
http://twitter.com/openlink

**Comment by joernhees** [comment]
well... until then you can build the .deb packages with docker and install them with apt like this:

```
docker run --rm -v ~/virtuoso_deb:/export/ joernhees/dpkg_build \
    https://github.com/openlink/virtuoso-opensource/archive/stable/7.tar.gz -j4
sudo "echo \"deb file:~/virtuoso_deb ./\" >> /etc/apt/sources.list.d/virtuoso_local_packages.list && apt-get update && apt-get install virtuoso-server"
```

The first command will build the .deb packages in a docker environment and copy the results to `~/virtuoso_deb`. The second command will add that folder to your apt sources and install the latest virtuoso-server version.

**Comment by oxymor0n** [comment]
@joernhees might not be a problem with your Dockerfile per se, but Make is throwing an error when building the latest version of Virtuoso Opensource (7.2.2):

```
/usr/bin/make -C binsrc/VirtuosoClient.Net clean
make[1]: *** No rule to make target 'clean'.  Stop.
make[1]: Entering directory '/tmp/build/virtuoso-opensource-stable-7/binsrc/VirtuosoClient.Net'
make[1]: Leaving directory '/tmp/build/virtuoso-opensource-stable-7/binsrc/VirtuosoClient.Net'
debian/rules:75: recipe for target 'cleanbuilddir/libvirtmono0-cil' failed
make: *** [cleanbuilddir/libvirtmono0-cil] Error 2
dpkg-buildpackage: error: debian/rules clean gave error exit status 2
```

**Comment by joernhees** [comment]
correct, the latest 7.2.2.1 release is broken... would be cool if there was a new release soon...

**Comment by openlink** [comment]
We are working on a new release which we hope to release in the next week or so.

**Comment by joernhees** [comment]
:+1:

**Comment by kjetilk** [comment]
Here's the Debian bug on getting a new Virtuoso version in: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=689230

I think it is very important to get more widespread adoption. It is not just about the ease of installing, you can have PPA's or similar for that. It is the commitment Debian makes that comes with long-term support, most importantly security support. For a SMB to adopt a piece of software into their product, you'd have to have someone following every piece you have in there for security issues. That is what Debian does for you, and it is a very, very valuable service. 

For the non-opensource version of Virtuoso, OpenLink does that job, but the ease of upgrades using the package management tools are still important.

I really hope you can make an effort to get Virtuoso 7 into Debian, and I think you would find it useful to have the same packaging available for your commercial version.

**Comment by andrewufrank** [comment]
i tried to use the OpenNode repository mentioned by Jan-Marcek abovve. i got the response:
````
GPG error: http://packages.comsode.eu/debian wheezy InRelease: The following signatures were invalid: 2CD0CD4558181DB983CEBA68691541B71090D51C
W: The repository 'http://packages.comsode.eu/debian wheezy InRelease' is not signed.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
````
could you please fix the key to make use of the repository save?  
in the repository there is a virtuoso 7.2 debian, but not for stretch - could somebody produce one - it would be helpful!
thank you

**Comment by ailin-nemui** [comment]
virtuoso doesn't build on debian stretch because of #583 #663

**Comment by HughWilliams** [comment]
The problem with the new openssl 1.1.0 version shipped with debian stretch is a known issue development hope to have a fix for in the upcoming virtuoso open source new release ...

**Comment by TheNavigateur** [comment]
@Jan-Marcek 

In performing your commands (as root) followed by `sudo aptitude install virtuoso-opensource` I got:

```
The following packages have unmet dependencies:

virtuoso-opensource-7 :

Depends: libmagickcore5 (>= 8:6.7.7.10) which is a virtual package and is not provided by any available package.

Depends: libmagickwand5 (>= 8:6.7.7.10) which is a virtual package and is not provided by any available package.
```

What is your solution that made this work as expected? I'm using Ubuntu Server 16.04. (I tried apt upgrade to no avail).

**Comment by mnieuwland** [comment]
I have the exact same error when installing virtuoso-opensource-7. Wasn't able to install the dependencies manually...... So I'm stuck.

**Comment by TheNavigateur** [comment]
@mnieuwland I just installed virtuoso opensource 7 from source code as indicated [here](https://github.com/openlink/virtuoso-opensource) .

I had [this](https://github.com/openlink/virtuoso-opensource/issues/736) issue (resolved as answered) and it's working fine now.

I did not resolve the issue with being able to do apt-get as in this thread.

I am fairly pleased because it means I can get the latest version easily whenever I want, even though the prospect of building from source code did seem daunting at first.

When I followed the instructions otherwise, it was no problem getting it to work normally.

**Comment by mnieuwland** [comment]
Thanks @TheNavigateur 

That worked for me. Does take a lot of extra time though....
